### ISSUE TITLE
Item37: TheadRAII的析构似乎并非异步

### ISSUE THREAD

**Issue by callMeBigBen** [kelthuzadx/EffectiveModernCppChinese#72]
在ThreadRAII的实例代码下提到：
> 这份代码中，我们选择在ThreadRAII的析构函数中异步执行join的动作

`~ThreadRAII() 
  {
    if (t.joinable()) {
      if (action == DtorAction::join) {
        t.join();
      } else {
        t.detach();
      }
    }
  }`

此处析构函数并非异步吧？
如果传入DtorAction为join，主线程应该会被ThreadRAII的析构阻塞，直到线程任务return。

**Comment by wendajiang** [comment]
确实，这里是翻译上的错误。
In this case, we’ve chosen to do a join on the asynchronously running thread in the ThreadRAII destructor, because, as we saw earlier, doing a detach could lead to some truly nightmarish debugging.

这是原文，已经fix，如果有更好的表述，可以帮忙PR下
